### PR TITLE
adding a statistics module

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

This PR adds a simple summary-statistics module to `pyam`, which acts as a wrapper for `pd.descibe()`. When initializing an instance of  `pyam.Statistics`, the user defines `groupby` and/or `filters` arguments, which will then be applied to all data added to the statistics instance.

This moves the tedious tasks of applying groupbys, filtering, structuring of data into the module, and formatting of the `describe()` output, so that the workflow can focus on "get some data and describe, get more data and describe, etc.", easily producing a coherent summary-statistics output.

The `pyam.Statistics.summarize()`offers simple selection of mean vs. median, interquartile vs. full  range, and rounding/formatting.

The PR also extends the `pyam.filter_by_meta()` function to accept `model` and `scenario` as filter arguments.

### CODE REVIEW

File: pyam/statistics.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        if isinstance(data, pd.Series):
+            if subheader is not None:
+                data.name = subheader
+            if data.name is None:
+                msg = '`data` must be named `pd.Series` or provide `subheader`'
+                raise ValueError(msg)
+            data = pd.DataFrame(data)
+
+        if self.groupby is not None:
+            args = dict(data=data, df=self.df, **self.groupby, join_meta=True)
```
[the remark this answers is not in the document]
Comment (danielhuppmann) [reply pyam/statistics.py]: well it may invalid, but it works...

**Comment by danielhuppmann** [comment]
@gidden, refactored the `Statistics.describe()` to `add()` as you suggested, made the `reindex()` function behave more like `pd.DataFrame.reindex()`, and some more clean-up.

**Comment by gidden** [comment]
ready for review?

**Comment by danielhuppmann** [comment]
just resolved the merge conflict, ready for review

**Comment by gidden** [comment]
Ok, lots in there, so I wasn't able to do a full in-depth review, but we can merge now to keep moving.

**Pull request merged**